feg/gateway/sbi/sbi_test.go

```diff
@@ -0,0 +1,165 @@
+func (pcf *MockPcf) GetSmPoliciesSmPolicyId(ctx echo.Context, smPolicyId string) error {
+	policy, found := pcf.Policies[smPolicyId]
+	if !found {
+		return ctx.NoContent(http.StatusNotFound)
+	}
+	return ctx.JSON(http.StatusOK, policy)
+}
+
+// PostSmPoliciesSmPolicyIdDelete handles POST /sm-policies/{smPolicyId}/delete
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdDelete(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+// PostSmPoliciesSmPolicyIdUpdate handles POST /sm-policies/{smPolicyId}/update
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdUpdate(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+func TestCreateSMPolicy(t *testing.T) {
+	// Start the server and wait till the server is started
+	mockPcf := startServer(t)
+
+	// Create new N7 client object
+	client, err := n7_client.NewClientWithResponses(fmt.Sprintf("http://%s", mockPcf.ListenAddr))
+
+	assert.NoError(t, err, "Client creation failed")
+
+	// Create a new SM Policy
+	ipAddr := sbi.Ipv4Addr(IP_ADDR1)
+	imsi := sbi.Supi(IMSI1)
+	body := n7_client.PostSmPoliciesJSONRequestBody{
+		Ipv4Address:  &ipAddr,
+		PduSessionId: 10,
+		Supi:         imsi,
+	}
+
+	resp, err := client.PostSmPolicies(context.Background(), body)
+	assert.NoError(t, err, "Failed to create SM Policies")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	// Get SM Policy given the policyId
+	policyResp, err := client.GetSmPoliciesSmPolicyIdWithResponse(context.Background(), POLICY_ID1)
+	assert.NoError(t, err, "Failed to get SM Policies")
+	assert.Equal(t, http.StatusOK, policyResp.StatusCode())
+	assert.Equal(t, sbi.Ipv4Addr(IP_ADDR1), *(policyResp.JSON200.Context.Ipv4Address))
+	assert.Equal(t, sbi.Supi(IMSI1), policyResp.JSON200.Context.Supi)
+}
+
+func startServer(t *testing.T) *MockPcf {
+	errChan := make(chan error)
+	eserver := echo.New()
+
+	mockPcf := NewMockPcf()
+	go func() {
+		n7.RegisterHandlers(eserver, mockPcf)
+
+		err := eserver.Start("localhost:0")
+		if err != nil {
+			errChan <- err
+		}
+	}()
+
+	addr, err := waitForTestServer(eserver, errChan)
+	assert.NoError(t, err)
+	mockPcf.ListenAddr = addr
+
+	return mockPcf
+}
+
+// waitForTestServer waits for the Echo server to be launched and returns the Listener Address.
+// There is a similar one in magma/orc8r/cloud/go/obsidian/access/tests but that uses a older
+// version of Echo server.
+func waitForTestServer(e *echo.Echo, errChan <-chan error) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 200*time.Millisecond)
+	defer cancel()
+
+	ticker := time.NewTicker(5 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return "", ctx.Err()
+		case <-ticker.C:
+			addr := e.ListenerAddr()
+			if addr != nil && strings.Contains(addr.String(), ":") {
+				// server started
+				return addr.String(), nil
+			}
+		case err := <-errChan:
+			if err == http.ErrServerClosed {
+				return "", nil
```

Comment:
Why is this function taking an error and not returning it? This seems to be hiding an error

Reply:
Ok looks fine, lets leave it as it is.
I am still not sure we need that case. It looks like that case may be useful for echo testing but not for us. But lets leave it as it is